brave/brave-core
# Show a rewards notification when Uphold wallet linking limit is reached

Resolves https://github.com/brave/brave-browser/issues/15028

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

### Scenario

* Given that a user has reached their Uphold linking limit
* When they connect their Uphold wallet to rewards
* Then a notification will be displayed in the rewards panel informing them that they've reached their device linking limit.
* And an entry will be recorded in the rewards database event log table.

### Test Steps

* Start the browser with a clean profile
* Navigate to `brave://rewards` and click `Verify Wallet`
* In the popup, click `Login`
* Log in to Uphold with an account that has reached its device linking limit
* Verify that a notification is displayed in the reward panel
* Navigate to `brave://rewards-internals` and select the `Event Log` tab
* Verify that a `device_limit_reached` entry appears in the event log

<img width="394" alt="Screen Shot 2021-03-31 at 10 34 09 AM" src="https://user-images.githubusercontent.com/5995084/113162001-e2f67e80-920c-11eb-8876-53ffdfd17e41.png">
